geopyspark/geotrellis/catalog.py

```diff
@@ -253,53 +83,68 @@ def read_value(layer_type,
         zdt (``datetime.datetime``): The time stamp of the tile if the data is spatial-temporal.
             This is represented as a ``datetime.datetime.`` instance.  The default value is,
             ``None``. If ``None``, then only the spatial area will be queried.
-        options (dict, optional): Additional parameters for reading the tile for specific backends.
-            The dictionary is only used for ``Cassandra`` and ``HBase``, no other backend requires
-            this to be set.
-        **kwargs: The optional parameters can also be set as keywords arguments. The keywords must
-            be in camel case. If both options and keywords are set, then the options will be used.
 
     Returns:
         :class:`~geopyspark.geotrellis.Tile`
     """
 
-    if not _in_bounds(layer_type, uri, layer_name, layer_zoom, col, row):
-        return None
+    if uri in _cached_stores:
+        store = _cached_stores[uri]
     else:
-        options = options or kwargs or {}
+        store = AttributeStore.build(uri)
+        _cached_stores[uri] = store
 
-        if zdt:
-            zdt = zdt.astimezone(pytz.utc).isoformat()
-        else:
-            zdt = ''
+    reader = ValueReader(uri, layer_name, layer_zoom, store)
+    return reader.read(col, row, zdt)
 
-        if uri not in _mapped_cached:
-            _construct_catalog(get_spark_context(), uri, options)
 
-        cached = _mapped_cached[uri]
+class ValueReader(object):
```

Comment:
Thats up for discussion. The really good thing about `ValueReader` being a class is that it holds an exact instance of `AttributeStore`. So if something goes wrong its a very direct link to what is likely causing a failure. 

It feels like this should be an option in the API that can expose more functionality aside from `read`. Like this would be the place to implement a way to `read_extent` and `read_thumbnail` and all the other functions that basically use Collections API.

It breaks from "everything is a function" we've strived for in the API but `AttributeStore` did that and actually ended up much more usable, this could be another case.